bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -1406,7 +1411,9 @@ void drawSelected(int itemIndex, GC gc, Rectangle bounds, int state ) {
 				}
 			}
 
-			// draw Text
+			// draw Text			
+			//Add some padding before text, if image is not visible
+			if(!parent.showSelectedImage) xDraw += INTERNAL_SPACING;
```

Comment:
is this not covered by the condition above already where spacing is added ?

Reply:
As far as I understand, the calculation above is for the size of the tab and this is for the placement of the text within it. Still I do not really like that the code is explained via so many comments (even though it is better compared having no explanation). I would prefer to have methods like `addTextPadding(...)` that consistently adds the padding whereever needed (i.e., both in the size calculation as well as in the text placement). That both avoids inconsistencies in the code and makes the code more comprehensible (and the comments obsolete).

Reply:
@sratz  As pointed out by @HeikoKlare, to add padding before the text, the width of the tab should also be increased otherwise the text would run out of the tab. Thus this change is to shift (pad) text by a standard width and same width size is added to the tab in the condition above.

@HeikoKlare We are working on extracting this logic into a separate method getTextPadding(...)  for uniform padding calculation when image is not visible.